nengo/params.py

```diff
@@ -497,18 +497,24 @@ class FrozenObject(object):
     multiple locations, compared, etc.
     """
 
+    # Order in which parameters have to be initialized.
+    # Missing parameters will be initialized last in an undefined order.
+    # This is needed for pickling and copying of Nengo objects when the
+    # parameter initialization order matters.
+    _param_init_order = []
+
     def __init__(self):
-        self._paramdict = {
-            k: v for k, v in inspect.getmembers(type(self))
-            if isinstance(v, Parameter) and not isinstance(v, ObsoleteParam)}
+        self._paramdict = collections.OrderedDict(
+            (k, v) for k, v in inspect.getmembers(type(self))
+            if isinstance(v, Parameter) and not isinstance(v, ObsoleteParam))
         for p in self._params:
             if not p.readonly:
                 msg = "All parameters of a FrozenObject must be readonly"
                 raise ReadonlyError(attr=p, obj=self, msg=msg)
 
     @property
     def _params(self):
-        return list(itervalues(self._paramdict))
+        return itervalues(self._paramdict)
```

Comment:
So the reason this was a list and not an iterator is because it's a property. When you do `thing.attribute` you do not expect to get back an iterator, conventionally iterators are only returned from methods (hence why we have `dict.values()` etc rather than `dict.values`). This is important because:

```python
param = self._params
for p in param:
   # this works fine
for p in param:
   # this now fails because the iterator is exhausted
```

I definitely wouldn't expect this behavior, as an attribute should just give me data. I would be more likely to expect something like this with `params = self._iterparams()`.

Reply:
Sure, makes sense!